File: domain/src/main/java/com/ahn/domain/setting/usecase/GetThemeModeUseCase.kt
package com.ahn.domain.setting.usecase

import com.ahn.domain.setting.model.ThemeMode
import com.ahn.domain.setting.repository.ThemeRepository
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class GetThemeModeUseCase @Inject constructor(private val repository: ThemeRepository) {
    operator fun invoke(): Flow<ThemeMode> {
        return repository.getThemeMode()
    }
}

File: ios/Packages/Domain/Sources/Domain/Setting/UseCase/GetThemeModeUseCase.swift
public struct GetThemeModeUseCase: Sendable {
    private let repository: any ThemeRepository

    public init(repository: any ThemeRepository) {
        self.repository = repository
    }

    public func callAsFunction() -> AsyncStream<ThemeMode> {
        repository.themeMode()
    }
}
